comicapi/genericmetadata.py

```diff
@@ -202,24 +212,122 @@ def get_clean_metadata(self, *attributes: str) -> GenericMetadata:
         new_md.__post_init__()
         return new_md
 
-    def overlay(self, new_md: GenericMetadata) -> None:
-        """Overlay a metadata object on this one
+    def assign_dedupe_new(self, new: list[Any] | set[Any], cur: str) -> None:
+        if isinstance(new, list):
+            for a in new:
+                for i, b in enumerate(getattr(self, cur)):
+                    if a.casefold() == b.casefold():
+                        getattr(self, cur).pop(i)
+                        getattr(self, cur).insert(i, a)
+                        break
+        if isinstance(new, set):
+            remove_set = set()
+            for a in new:
+                for b in getattr(self, cur):
+                    if a.casefold() == b.casefold():
+                        remove_set.add(b)
+                        break
+            getattr(self, cur).difference_update(remove_set)
+            getattr(self, cur).update(new)
+
+    def assign_dedupe_cur(self, new: list[Any] | set[Any], cur: str) -> None:
+        if isinstance(new, list):
+            remove_ids = set()
+            for a in getattr(self, cur):
+                for i, b in enumerate(new):
+                    if a.casefold() == b.casefold():
+                        remove_ids.add(i)
+                        break
+            for r_id in remove_ids:
+                new.pop(r_id)
+            getattr(self, cur).extend(new)
+
+        if isinstance(new, set):
+            remove_set = set()
+            for a in getattr(self, cur):
+                for b in new:
+                    if a.casefold() == b.casefold():
+                        remove_set.add(b)
+                        break
+            new.difference_update(remove_set)
+            getattr(self, cur).update(new)
+
+    def assign_original(self, cur: str, new: Any) -> None:
+        if new is not None:
+            if isinstance(new, str) and len(new) == 0:
+                if isinstance(getattr(self, cur), (list, set)):
+                    getattr(self, cur).clear()
+                else:
+                    setattr(self, cur, None)
+            elif isinstance(new, (list, set)) and len(new) == 0:
+                pass
+            else:
+                setattr(self, cur, new)
+
+    def assign_overwrite(self, cur: str, new: Any) -> None:
```

Comment:
If everything is being overwritten isn't it just making a copy of the new metadata object?

Reply:
I don't know if it was supposed to do something else originally but yes, it will now overwrite everything. I think this should then replace the `Clear all existing metadata before import...` so everything is in one location/setting?

Reply:
We can look at that after this pr